core/soong_gen.go

```diff
@@ -370,26 +370,21 @@ func (m *genBackend) AndroidMkEntries() []android.AndroidMkEntries {
 	}}
 }
 
-func (gc *generateCommon) getHostBinModule(mctx android.TopDownMutatorContext) (hostBin *binary) {
-	var hostBinModule android.Module
+func (gc *generateCommon) getHostBinModule(mctx android.TopDownMutatorContext) (hostBin android.Module) {
 	mctx.VisitDirectDepsWithTag(hostToolBinTag, func(m android.Module) {
-		hostBinModule = m
+		hostBin = m
 	})
-	if hostBinModule == nil {
+	if hostBin == nil {
 		panic(fmt.Errorf("Could not find module specified by `host_bin: %v`", proptools.String(gc.Properties.Host_bin)))
 	}
-	bin, ok := hostBinModule.(*binary)
```

Comment:
Useful to keep this check too?

Reply:
Yep - although I've just spotted that the panic() inside the `if` incorrectly references `bin`, which would be Nil if the check failed. So I'll leave the check, but fix that.